libs/langchain/langchain/vectorstores/mongodb_atlas.py

```diff
@@ -159,24 +198,64 @@ def _similarity_search_with_score(
         }
         if pre_filter:
             knn_beta["filter"] = pre_filter
-        pipeline = [
-            {
-                "$search": {
-                    "index": self._index_name,
-                    "knnBeta": knn_beta,
-                }
-            },
-            {"$set": {"score": {"$meta": "searchScore"}}},
-        ]
-        if post_filter_pipeline is not None:
-            pipeline.extend(post_filter_pipeline)
-        cursor = self._collection.aggregate(pipeline)  # type: ignore[arg-type]
-        docs = []
-        for res in cursor:
-            text = res.pop(self._text_key)
-            score = res.pop("score")
-            docs.append((Document(page_content=text, metadata=res), score))
-        return docs
+        query = {
+            "$search": {
+                "index": self._index_name,
+                "knnBeta": knn_beta,
+            }
+        }
+
+        return self._similarity_search_query(query, post_filter_pipeline)
+
+    def _similarity_search_query_vectorsearch(
+        self,
+        embedding: List[float],
+        k: int = 4,
+        pre_filter: Optional[dict] = None,
+        post_filter_pipeline: Optional[List[Dict]] = None,
+    ) -> List[Tuple[Document, float]]:
+        params = {
+            "queryVector": embedding,
+            "path": self._embedding_key,
+            "numCandidates": 100,
```

Comment:
I saw in the docs we used 100 and I was a little unclear on the purpose of this key in those docs. Any notion from the vector search devs on how much a developer would need this param tuned?

Reply:
Whoops this is supposed to be `k`, not 100. From my understanding having both `limit` and `numCandidates` be `k` is the equivalent of what `$search` does.